File: backend/src/main/java/com/backend/api/user/dto/response/UserLoginResponse.java
package com.backend.api.user.dto.response;

import com.backend.domain.user.entity.User;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

public record UserLoginResponse(
        @Schema(description = "사용자 ID", example = "1")
        Long id,

        @Schema(description = "사용자 이메일", example = "user@example.com")
        String email,

        @Schema(description = "사용자 이름", example = "홍길동")
        String name,

        @Schema(description = "사용자 닉네임", example = "spring_dev")
        String nickname,

        @Schema(description = "사용자 나이", example = "25")
        Integer age,

        @Schema(description = "사용자 GitHub 프로필 URL", example = "https://github.com/user")
        String github,

        @Schema(description = "사용자 프로필 이미지 URL", example = "https://example.com/profile.jpg")
        String image,

        @Schema(description = "사용자 권한", example = "USER")
        String role,

        @Schema(description = "생성일", example = "2025-10-13T10:15:30")
        LocalDateTime createDate,

        @Schema(description = "수정일", example = "2025-10-13T10:20:00")
        LocalDateTime modifyDate,

        @Schema(description = "accessToken", example = "accessToken")
        String accessToken,

        @Schema(description = "refreshToken", example = "refreshToken")
        String refreshToken


) {
    public static UserLoginResponse from(User user, String accessToken, String refreshToken) {
        return new UserLoginResponse(
                user.getId(),
                user.getEmail(),
                user.getName(),
                user.getNickname(),
                user.getAge(),
                user.getGithub(),
                user.getImage(),
                user.getRole().name(),
                user.getCreateDate(),
                user.getModifyDate(),
                accessToken,
                refreshToken

        );
    }

    public static UserLoginResponse from(User user) {
        return new UserLoginResponse(
                user.getId(),
                user.getEmail(),
                user.getName(),
                user.getNickname(),
                user.getAge(),
                user.getGithub(),
                user.getImage(),
                user.getRole().name(),
                user.getCreateDate(),
                user.getModifyDate(),
                null,
                null
        );
    }
}

File: backend/src/main/java/com/backend/api/user/dto/response/UserLoginResponse.kt
package com.backend.api.user.dto.response

import com.backend.domain.user.entity.User
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

data class UserLoginResponse(
    @field:Schema(description = "사용자 ID", example = "1")
    val id: Long,

    @field:Schema(description = "사용자 이메일", example = "user@example.com")
     val email: String,

    @field:Schema(description = "사용자 이름", example = "홍길동")
    val name: String,

    @field:Schema(description = "사용자 닉네임", example = "spring_dev")
    val nickname: String,

    @field:Schema(description = "사용자 나이", example = "25")
    val age: Int,

    @field:Schema(description = "사용자 GitHub 프로필 URL", example = "https://github.com/user")
    val github: String,

    @field:Schema(description = "사용자 프로필 이미지 URL", example = "https://example.com/profile.jpg")
    val image: String?,

    @field:Schema(description = "사용자 권한", example = "USER")
    val role: String,

    @field:Schema(description = "생성일", example = "2025-10-13T10:15:30")
    val createDate: LocalDateTime,

    @field:Schema(description = "수정일", example = "2025-10-13T10:20:00")
    val modifyDate: LocalDateTime,

    @field:Schema(description = "accessToken", example = "accessToken")
    val accessToken: String?,
    @field:Schema(description = "refreshToken", example = "refreshToken")
    val refreshToken: String?


) {
    companion object {
        fun from(user: User, accessToken: String, refreshToken: String): UserLoginResponse {
            return UserLoginResponse(
                user.id,
                user.email,
                user.name,
                user.nickname,
                user.age,
                user.github,
                user.image,
                user.role.name,
                user.createDate,
                user.modifyDate,
                accessToken,
                refreshToken
            )
        }

        fun from(user: User): UserLoginResponse {
            return UserLoginResponse(
                user.id,
                user.email,
                user.name,
                user.nickname,
                user.age,
                user.github,
                user.image,
                user.role.name,
                user.createDate,
                user.modifyDate,
                null,
                null
            )
        }
    }
}
